Islandora/islandora
# Add an "input args" parameter when defining derivative actions.

* Other Relevant Links:

Related PRs:
 * https://github.com/Islandora/Alpaca/pull/91
 * https://github.com/Islandora/Crayfish/pull/172

# What does this Pull Request do?

This PR adds an "input args" parameter in derivative actions, to be used to pass additional arguments when generating image derivatives.  This fields is only exposed in the UI for image derivatives (not for other derivative types), as images are the only use case which needs this additional field.

# Background/Motivation

I wanted to use the "density" parameter in imagemagick, but this parameter only works if applied BEFORE the input stream. According to the man page (https://manpages.ubuntu.com/manpages/trusty/man1/convert.im6.1.html) there are several options which are only valid as so-called "input options".  The addition of this field makes it possible to use these parameters to adjust the conversion.  In particular, the density parameter makes it possible to generate higher-resolution JPGs from a PDF source file.

# What's new?

One field is added to the "Edit action" page for image derivatives.  

The field should not show for other derivative types (I think I got them all - I followed existing code to remove the field from their configuration forms).

Until the related Alpaca PR is merged, the addition of this field will cause alpaca to spew errors (about the new unknown field).

I think this means some kind of version constraint will need to be applied, but I don't know exactly what/how.

# How should this be tested?

It is easier to test these PRs together.  See in-depth test plan in https://github.com/Islandora/Alpaca/pull/91

# Documentation Status

* I have added some inline documentation into the Action configuration page (links to imagemagick documentation).

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/committers
